src/test/java/unit/DataSourceReferenced.java

```diff
@@ -181,6 +183,42 @@ public void resolve() throws IOException {
         "resource", "delete", "--name=" + DATA_SOURCE_RESOURCE_NAME, "--quiet");
   }
 
+  @Test
+  @DisplayName("Test running terra app execute env in a workspace with data source")
+  void appExecuteEnv() throws IOException {
+    workspaceCreator.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getUserFacingId());
+
+    // `terra app execute env`
+    Result result = TestCommand.runAndGetResultExpectSuccess("app", "execute", "env");
+    String stdOut = result.stdOut;
+    assertDataSourceResourcesEnv(stdOut);
+
+    // `terra resource delete --name=$name`
+    TestCommand.runCommandExpectSuccess(
+        "resource", "delete", "--name=" + DATA_SOURCE_RESOURCE_NAME, "--quiet");
+  }
+
+  private void assertDataSourceResourcesEnv(String stdOut) {
```

Comment:
nit - The three assert methods are in reverse order compared to the tests. Can we either move each helper method right below the test that uses it, or put them in the order they're used?